testsuite/smoke-test/src/rosetta.rs

```diff
@@ -713,11 +751,199 @@ async fn test_transfer() {
     */
 }
 
+#[ignore]
+#[tokio::test]
+async fn test_delegation_pool_operations() {
+    let (mut swarm, cli, _faucet, rosetta_client) = setup_test(1, 3).await;
+
+    let _account_1 = cli.account_id(0);
+    cli.fund_account(0, Some(100000000)).await.unwrap();
+    let private_key_0 = cli.private_key(0);
+
+    let chain_id = swarm.chain_id();
+    let validator = swarm.validators().next().unwrap();
+    let rest_client = validator.rest_client();
+    let _request = NetworkRequest {
+        network_identifier: NetworkIdentifier::from(chain_id),
+    };
+    let network_identifier = chain_id.into();
+
+    // let chain_id = swarm.chain_id();
+    let root_address = swarm.aptos_public_info().root_account().address();
+    let root_sequence_number = swarm
+        .aptos_public_info()
+        .client()
+        .get_account_bcs(root_address)
+        .await
+        .unwrap()
+        .into_inner()
+        .sequence_number();
+    *swarm
+        .aptos_public_info()
+        .root_account()
+        .sequence_number_mut() = root_sequence_number;
+
+    let mut account_4 = swarm
+        .aptos_public_info()
+        .create_and_fund_user_account(1_000_000_000_000_000)
+        .await
+        .unwrap();
+
+    let mut account_5 = swarm
+        .aptos_public_info()
+        .create_and_fund_user_account(1_000_000_000_000_000)
+        .await
+        .unwrap();
+
+    let res = create_delegation_pool(&swarm.aptos_public_info(), &mut account_4, 10, 1).await;
+
+    let (tx, _) = res.into_parts();
+    let tx_serialized = json!(tx);
+    let mut pool_address_str = "";
+
+    if let Some(changes) = tx_serialized["changes"].as_array() {
+        for change in changes {
+            if change["data"]["type"] == "0x1::delegation_pool::DelegationPool" {
+                pool_address_str = change["address"].as_str().unwrap();
+                break;
+            }
+        }
+    }
```

Comment:
This you can use `serde_json::from_string` to convert it into a real type, which means you don't have to deal with the bracket notation